plugins/extractors/merlin/internal/merlin/merlin_client.go

```diff
@@ -149,15 +148,15 @@ func (c Client) exec(req *http.Request, result interface{}) error {
 
 func authenticatedClient(ctx context.Context, serviceAccountJSON []byte, scopes ...string) (*http.Client, error) {
 	if len(serviceAccountJSON) == 0 {
-		return google.DefaultClient(ctx, scopes...)
+		return http.DefaultClient, nil
 	}
 
-	creds, err := google.CredentialsFromJSON(ctx, serviceAccountJSON, authScopes...)
+	creds, err := idtoken.NewClient(ctx, authScope, idtoken.WithCredentialsJSON(serviceAccountJSON))
```

Comment:
second argument is audience, not auth scope, 
merlin have python sdk, try to use that target audience (`sdk.caraml`)
https://github.com/caraml-dev/merlin/blob/367a4ecb4a12dfd380cc776197ca9f3b510195c6/python/sdk/merlin/client.py#L56

let's create a constant 

```suggestion
	client, err := idtoken.NewClient(ctx, audience, idtoken.WithCredentialsJSON(serviceAccountJSON))
```